libkbfs/disk_block_metadata_store.go

```diff
@@ -80,135 +80,146 @@ func newDiskBlockMetadataStore(
 		}
 		return value, nil
 	default:
-		c.log.CWarningf(ctx, "getMetadata error: %v", err)
-		return DiskBlockMetadataValue{}, ldberrors.ErrNotFound
+		s.log.CWarningf(ctx, "GetMetadata error: %v", err)
+		return BlockMetadataValue{}, ldberrors.ErrNotFound
 	}
 }
 
-type blockMetadataUpdater func(*DiskBlockMetadataValue) error
-
-func (c *diskBlockMetadataStore) setMetadata(ctx context.Context,
-	blockID kbfsblock.ID, updater blockMetadataUpdater) error {
+// UpdateMetadata implements the BlockMetadataStore interface.
+func (s *diskBlockMetadataStore) UpdateMetadata(ctx context.Context,
+	blockID kbfsblock.ID, updater BlockMetadataUpdater) error {
 	bid := blockID.Bytes()
 
-	c.lock.Lock()
-	defer c.lock.Unlock()
+	s.lock.Lock()
+	defer s.lock.Unlock()
 
 	select {
-	case <-c.shutdownCh:
-		return ErrDiskBlockMetadataStoreShutdown{}
+	case <-s.shutdownCh:
+		return ErrBlockMetadataStoreShutdown{}
 	default:
 	}
 
-	var value DiskBlockMetadataValue
-	encoded, err := c.db.Get(bid, nil)
+	var value BlockMetadataValue
+	encoded, err := s.db.Get(bid, nil)
 	switch errors.Cause(err) {
 	case ldberrors.ErrNotFound:
 	case nil:
-		if err = c.config.Codec().Decode(encoded, &value); err != nil {
-			c.log.CWarningf(ctx, "decoding block metadata error: %v", err)
+		if err = s.config.Codec().Decode(encoded, &value); err != nil {
+			s.log.CWarningf(ctx, "decoding block metadata error: %v", err)
 		}
 	default:
-		c.log.CWarningf(ctx, "getMetadata error: %v", err)
+		s.log.CWarningf(ctx, "GetMetadata error: %v", err)
 	}
 
 	if err = updater(&value); err != nil {
 		return err
 	}
 
-	if encoded, err = c.config.Codec().Encode(value); err != nil {
+	if encoded, err = s.config.Codec().Encode(value); err != nil {
 		return err
 	}
-	if err = c.db.PutWithMeter(bid, encoded, c.putMeter); err != nil {
+	if err = s.db.PutWithMeter(bid, encoded, s.putMeter); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// GetXattr looks for and returns the Xattr value of xattrType for blockID if
-// it's found, and ldberrors.ErrNotFound if it's not found.
-func (c *diskBlockMetadataStore) GetXattr(ctx context.Context,
+// xattrStore is a wrapper around BlockMetadataStore that handles xattr
+// values.
+type xattrStore struct {
+	store BlockMetadataStore
+}
+
+// NewXattrStoreFromBlockMetadataStore returns a XattrStore which is a wrapper
+// around the passed in store.
+func NewXattrStoreFromBlockMetadataStore(store BlockMetadataStore) XattrStore {
+	return xattrStore{store: store}
+}
+
+var _ XattrStore = (*xattrStore)(nil)
+
+// GetXattr implements the XattrStore interface.
+func (s xattrStore) GetXattr(ctx context.Context,
 	blockID kbfsblock.ID, xattrType XattrType) ([]byte, error) {
-	blockMetadata, err := c.getMetadata(ctx, blockID)
+	blockMetadata, err := s.store.GetMetadata(ctx, blockID)
 	if err != nil {
 		return nil, err
 	}
 
 	v, ok := blockMetadata.Xattr[xattrType]
 	if !ok {
-		c.missMeter.Mark(1)
+		s.store.MarkMiss(1)
```

Comment:
This should be tracking its own miss meter, rather than mutating the `MetadataStore` miss meter. They represent different information.

Reply:
GOod point!